### PR TITLE
Feature: TCOTA-513 - Assignments Tab

### PR DESCRIPTION
**Pull request by rmathis**

### CODE REVIEW

**Review by machristie — CHANGES_REQUESTED** [review]
Thanks @rmathis. I've requested a few small changes in my review.

File: src/main/javascript/src/views/ExperimentAssignments.vue [hunk 1]
```diff
@@ -0,0 +1,531 @@
...
+};
+</script>
+
+<style lang="scss">
+.v-data-table__wrapper {
+  // border-radius: 0;
+  table tbody {
+  }
+}
+
```
[the remark this answers is not in the document]
Comment (rmathis) [reply src/main/javascript/src/views/ExperimentAssignments.vue]: Fixed

File: src/main/javascript/src/views/ExperimentAssignments.vue [hunk 2]
```diff
@@ -0,0 +1,531 @@
...
+                        :headers="treatmentHeaders"
+                        :items="item.treatments"
+                        hide-default-header
+                        hide-default-footer
+                        item-key="title"
+                        class="grey lighten-5"
+                      >
+                        <!-- eslint-disable-next-line -->
+                        <template v-slot:item.title="{ item }">
+                          {{ item.assessmentDto.title }}
```
[the remark this answers is not in the document]
Comment (rmathis) [reply src/main/javascript/src/views/ExperimentAssignments.vue]: Fixed

File: src/main/javascript/src/views/ExperimentAssignments.vue [hunk 3]
```diff
@@ -0,0 +1,531 @@
...
+  <div>
+    <v-container class="px-0" v-if="experiment">
+      <v-row>
+        <v-col cols="12">
+          <v-divider class=""></v-divider>
+          <v-tabs v-model="tab" elevation="0" show-arrows>
+            <v-tab v-for="(exposure, index) in exposures" :key="index">
+              <div class="d-flex flex-column align-start py-1">
+                <div class="">Set {{ index + 1 }}</div>
+                <div class="d-block red--text mt-4">
```
[the remark this answers is not in the document]
Comment (rmathis) [reply src/main/javascript/src/views/ExperimentAssignments.vue]: Fixed

File: src/main/javascript/src/views/ExperimentAssignments.vue [hunk 4]
```diff
@@ -0,0 +1,531 @@
...
+    },
+    saveExit() {
+      this.$router.push({ name: "Home" });
+    },
+    getAssignmentsForExposure(exp) {
+      return this.assignments
+        .filter((a) => a.exposureId === exp.exposureId)
+        .sort((a, b) => a.assignmentOrder - b.assignmentOrder);
+    },
+    async exportData() {
```
[the remark this answers is not in the document]
Comment (rmathis) [reply src/main/javascript/src/views/ExperimentAssignments.vue]: Fixed

File: src/main/javascript/src/views/ExperimentAssignments.vue [hunk 5]
```diff
@@ -0,0 +1,531 @@
...
+          editSection: "AssignmentExposureSets",
+          type: "assignments",
+        },
+      ];
+    },
+    conditionCount() {
+      return `${this.experiment.conditions.length} condition${
+        this.experiment.conditions.length > 1 ? "s" : ""
+      }`;
+    },
```
[the remark this answers is not in the document]
Comment (rmathis) [reply src/main/javascript/src/views/ExperimentAssignments.vue]: Fixed

File: src/main/javascript/src/views/ExperimentAssignments.vue [hunk 6]
```diff
@@ -0,0 +1,531 @@
...
+        c.conditionId,
+        this.assignments[0].assignmentId,
+      ]);
+      this.conditionTreatments[c.conditionId] = t?.data;
+    }
+    this.getAssignmentDetails();
+    await this.getZip(this.experiment.experimentId);
+    if (this.experiment.participationType === "CONSENT") {
+      await this.getConsentFile(this.experiment.experimentId);
+    }
```
[the remark this answers is not in the document]
Comment (rmathis) [reply src/main/javascript/src/views/ExperimentAssignments.vue]: Fixed

**Review by machristie — APPROVED** [review]
Looks good, thanks @rmathis

**Pull request merged**